app/controllers/aviti_sample_sheet/sample_sheet_generator.rb

```diff
@@ -102,6 +102,35 @@ def group_samples_by_identity
       grouped_samples
     end
 
+    # Determines the length of sample indexes used in the batch.
+    # This method iterates through grouped samples and retrieves the length of the first non-nil index (tag1 or tag2).
+    # It assumes all sample in the batch contains indexes and are of uniform length.
+    def sample_index_length
+      group_samples_by_identity.each_value do |group|
+        length = group[:tag1]&.length || group[:tag2]&.length
+        return length if length
+      end
+    end
+
+    #
+    # In Aviti runs, all sample indexes are expected to be of the same length.
+    # Adjusts the PhiX control indexes to match the sample index length used in the batch.
+    # The PSD-assisted software supports sample index lengths of either 8 bp or 10 bp.
+    # The default PhiX control indexes are 10 bp long.
+    #
+    # This method ensures that PhiX control sample indexes are truncated to match the actual
+    # sample index length in the current batch to maintain compatibility and prevent sequencing issues.
+    def adjusted_phix_indexes
+      PHIX_SECTION.map do |row|
+        if row.length > 2
+          row = row.dup
+          row[1] = row[1][0, sample_index_length] if row[1]
```

Comment:
If the index1 is ATGTCGCTAG and index length is 8, the index is changed to ATGTCGCT. Is this matching the reality? Do they have different controls with shorter indexes?